File: password-validator-python/password_validator.py
import re

def validate_password(password):
    # Periksa panjang minimal 15 karakter
    if len(password) < 15:
        return False

    # Definisikan kategori karakter
    has_uppercase = bool(re.search(r'[A-Z]', password))
    has_lowercase = bool(re.search(r'[a-z]', password))
    has_digit = bool(re.search(r'[0-9]', password))
    has_special_char = bool(re.search(r'[!@#$%^&*]', password))

    # Hitung jumlah kategori yang terpenuhi
    categories_met = sum([
        has_uppercase,
        has_lowercase,
        has_digit,
        has_special_char
    ])

    # Kembalikan True jika minimal 2 kategori terpenuhi
    return categories_met >= 2

File: password-validator-node/passwordValidator.js
function validatePassword(password) {
  // Periksa panjang minimal 15 karakter
  if (password.length < 15) {
    return false;
  }

  // Definisikan kategori karakter
  const hasUpperCase = /[A-Z]/.test(password);
  const hasLowerCase = /[a-z]/.test(password);
  const hasDigit = /[0-9]/.test(password);
  const hasSpecialChar = /[!@#$%^&*]/.test(password);

  // Hitung jumlah kategori yang terpenuhi
  const categoriesMet = [
    hasUpperCase,
    hasLowerCase,
    hasDigit,
    hasSpecialChar
  ].filter(Boolean).length;

  // Kembalikan true jika minimal 2 kategori terpenuhi
  return categoriesMet >= 2;
}

module.exports = { validatePassword };